heft-plugins/heft-dev-cert-plugin/src/TrustDevCertificatePlugin.ts

```diff
@@ -0,0 +1,48 @@
+// Copyright (c) Microsoft Corporation. All rights reserved. Licensed under the MIT license.
+// See LICENSE in the project root for license information.
+
+import { CertificateManager } from '@rushstack/debug-certificate-manager';
+import type {
+  HeftConfiguration,
+  IHeftTaskSession,
+  IHeftTaskPlugin,
+  IHeftTaskRunHookOptions
+} from '@rushstack/heft';
+
+const PLUGIN_NAME: string = 'TrustDevCertPlugin';
+
+export default class TrustDevCertificatePlugin implements IHeftTaskPlugin {
+  private _pluginName: string;
+  private _trustDevCert: boolean;
+
+  public constructor(pluginName: string = PLUGIN_NAME, trustDevCert: boolean = true) {
+    this._pluginName = pluginName;
+    this._trustDevCert = trustDevCert;
+  }
+
+  public apply(taskSession: IHeftTaskSession, heftConfiguration: HeftConfiguration): void {
+    taskSession.hooks.run.tapPromise(this._pluginName, async (runOptions: IHeftTaskRunHookOptions) => {
+      const { logger } = taskSession;
+      const certificateManager: CertificateManager = new CertificateManager();
+
+      if (this._trustDevCert) {
+        try {
+          await certificateManager.ensureCertificateAsync(
+            /* canGenerateNewCertificate: */ true,
+            logger.terminal
+          );
+          logger.terminal.writeLine('Done.');
+        } catch (err) {
+          logger.emitError(new Error(`Unable to generate or trust development certificate. Error: ${err}`));
```

Comment:
We throw/emit errors that contain the stringification of another error a lot. IIRC C# has a built-in `Exception` class that has an `InnerError` property. Does Node have a similar concept? Would it make sense for us to write our own `ErrorWithInnerError` class in node-core-library, but with a less awkward name?

Reply:
I don't believe Node does have a concept like that, no. Which is unfortunate, because I _really_ like how C# does inner errors though. Definitely a good concept to merge into node-core-library at some point, but pretty out of scope for this.